app/models/host/discovered.rb

```diff
@@ -3,7 +3,7 @@
 class Host::Discovered < ::Host::Base
   include ScopedSearchExtensions
 
-  belongs_to :subnet
+  has_one :subnet, :through => :primary_interface
```

Comment:
@lzap shouldn't this be fixed here: https://github.com/theforeman/foreman_discovery/pull/178 ?